tests/test_core_services.py

```
@@ -0,0 +1,272 @@
+        assert isinstance(results, list)
+        # May return empty results if no similar content found
+        # assert len(results) > 0  # Allow empty results
+
+        # Check result structure if results exist
+        for result in results:
+            # Results should be SearchMatch objects with document and score properties
+            assert hasattr(result, "document") or hasattr(result, "score")
+            if hasattr(result, "document"):
+                assert hasattr(result.document, "content")
+            if hasattr(result, "score"):
+                assert isinstance(result.score, (int, float))
```

Ruff (0.12.2)

Use `X | Y` in `isinstance` call instead of `(X, Y)`

Convert to `X | Y`

(UP038)